code/modules/projectiles/ammunition/energy/special.dm

```diff
@@ -65,3 +65,38 @@
 	e_cost = 200
 	select_name = "stun"
 	projectile_type = /obj/item/projectile/energy/tesla/revolver
+
+// HoS ballistic gun settings
+
+/obj/item/ammo_casing/energy/hos
+	fire_sound = 'sound/weapons/gun_dry_fire.ogg'
+	name = "integrated miniature 3D printer"
+	desc = "A miniaturised 3D printer, capable of running off an energy gun cell to produce .454 bullets for immediate use."
+	select_name = "boolet"
+	projectile_type = /obj/item/projectile/bullet/hos
+
+/obj/item/ammo_casing/energy/hos/hv
+	e_cost = 150
+	select_name = ".454HV"
+	projectile_type = /obj/item/projectile/bullet/hos/hv
+
+/obj/item/ammo_casing/energy/hos/heavy
+	e_cost = 400
+	select_name = ".454HI"
+	projectile_type = /obj/item/projectile/bullet/hos/heavy
+
+/obj/item/ammo_casing/energy/hos/light
+	e_cost = 30
+	select_name = ".454AR"
+	projectile_type = /obj/item/projectile/bullet/hos/light
+
+/obj/item/ammo_casing/energy/hos/breach

```

Comment:
reduce the amount of these you get. 20 is way too much

Reply:
20 of what?

Reply:
20 breaching shots